src/hera/workflows/script.py

```diff
@@ -156,66 +118,74 @@ def _build_inputs(self) -> Optional[ModelInputs]:
+        return self.constructor.transform_template(
+            self,
+            _ModelTemplate(
+                active_deadline_seconds=self.active_deadline_seconds,
+                affinity=self.affinity,
+                archive_location=self.archive_location,
+                automount_service_account_token=self.automount_service_account_token,
+                daemon=self.daemon,
+                executor=self.executor,
+                fail_fast=self.fail_fast,
+                host_aliases=self.host_aliases,
+                init_containers=self.init_containers,
+                inputs=self._build_inputs(),
+                memoize=self.memoize,
+                metadata=self._build_metadata(),
+                metrics=self.metrics,
+                name=self.name,
+                node_selector=self.node_selector,
+                outputs=self._build_outputs(),
+                parallelism=self.parallelism,
+                plugin=self.plugin,
+                pod_spec_patch=self.pod_spec_patch,
+                priority=self.priority,
+                priority_class_name=self.priority_class_name,
+                resource=self._build_resources(),
+                retry_strategy=self.retry_strategy,
+                scheduler_name=self.scheduler_name,
+                script=self._build_script(),
+                security_context=self.pod_security_context,
+                service_account_name=self.service_account_name,
+                sidecars=self.sidecars,
+                synchronization=self.synchronization,
+                timeout=self.timeout,
+                tolerations=self.tolerations,
+                volumes=self._build_volumes(),
+            ),
         )
 
     def _build_script(self) -> _ModelScriptTemplate:
-        return _ModelScriptTemplate(
-            args=self.args,
-            command=self.command,
-            env=self._build_env(),
-            env_from=self._build_env_from(),
-            image=self.image,
-            image_pull_policy=self._build_image_pull_policy(),
-            lifecycle=self.lifecycle,
-            liveness_probe=self.liveness_probe,
-            name=self.container_name,
-            ports=self.ports,
-            readiness_probe=self.readiness_probe,
-            resources=self._build_resources(),
-            security_context=self.security_context,
-            source=self._build_source(),
-            startup_probe=self.startup_probe,
-            stdin=self.stdin,
-            stdin_once=self.stdin_once,
-            termination_message_path=self.termination_message_path,
-            termination_message_policy=self.termination_message_policy,
-            tty=self.tty,
-            volume_devices=self.volume_devices,
-            volume_mounts=self._build_volume_mounts(),
-            working_dir=self.working_dir,
+        assert isinstance(self.constructor, ScriptConstructor)
+        return self.constructor.transform_script_template(
```

Comment:
Some additional hooks to allow script constructors to modify things even further if they wish to influence other params.